modules/imgproc/src/imgwarp.cpp

```diff
@@ -759,6 +759,14 @@ static void remapBilinear( const Mat& _src, Mat& _dst, const Mat& _xy,
             {
                 if( borderType == BORDER_TRANSPARENT && cn != 3 )
                 {
+                    int sx = XY[dx*2], sy = XY[dx*2+1];
+                    if ((unsigned)sx == width1 || (unsigned)sy == height1)
```

Comment:
BTW, we have strange code above:

```
#if CV_SIMD128
    if( _src.type() == CV_8UC3 )
        width1 = std::max(ssize.width-2, 0);
#endif
```

Why do we guard processing of common variables in SIMD mode only?

/cc @vpisarev That commit is very old: https://github.com/opencv/opencv/blame/fdd83e5027cab3dcbc6acb321ca9c294b90f17e1/modules/imgproc/src/imgwarp.cpp#L663

---

Code condition is strange too: `borderType == BORDER_TRANSPARENT && cn != 3`. Why is `cn` used here?

Reply:
> Code condition is strange too: borderType == BORDER_TRANSPARENT && cn != 3. Why is cn used here?

Removed that limitation and modified test correspondingly